### DISCUSSION TITLE
Performance Comparison: Flutter Desktop vs. Electron

### DISCUSSION THREAD

**Discussion by macbugs** [vercel/hyper#6918] [Ideas]
> - [**# Hyper is an Electron-based terminal**](https://hyper.is/)
> - [**_"Hyper 3 is finally out! The primary focus for this release is performance."_**](https://hyper.is/blog)
> 
___
- Performance Comparison: Flutter Desktop vs. Electron
_**["Flutter uses fewer resources in all the categories: CPU, GPU, Memory, and Energy.  The stand-out result, however, is the massive amount of memory used in the Electron version.  Roughly 2.2 GB compared to Flutter at ~170 MB."](https://getstream.io/blog/flutter-desktop-vs-electron/)**_

- [_"A simple Flutter app took 50 megabytes on disk and 25 megabytes of memory on Linux. Electron usually takes twice of these numbers."_](https://medium.com/swlh/goodbye-electron-welcome-flutter-22b3dc10d2f3)

- [Google is partnering with the Ubuntu Desktop Team at Canonical to bring Flutter apps to Linux.](https://www.xda-developers.com/google-partners-canonical-bring-flutter-apps-linux-ubuntu-snap-store/)

- [Why It’s Time To Consider Flutter Over Electron for the Desktop](https://betterprogramming.pub/why-its-time-to-consider-flutter-over-electron-for-the-desktop-213bba8930fa)

**Comment by hamzaaouni** [comment]
Flutter Desktop offers superior performance, development efficiency, and user interface capabilities compared to Electron

**Comment by tripplilley** [comment]
```shell-session
[tlilley:~/src/hyper] <system> canary ± scc .
───────────────────────────────────────────────────────────────────────────────
Language                 Files     Lines   Blanks  Comments     Code Complexity
───────────────────────────────────────────────────────────────────────────────
TypeScript                  96     10044     1122       602     8320       1280
JSON                        14       785        0         0      785          0
JavaScript                   9    151214    16741     13396   121077      16559
TypeScript Typings           6       665       48       135      482        134
Markdown                     5       388       81         0      307          0
YAML                         4       236       10        30      196          0
BASH                         3        87        9        15       63         15
gitignore                    2        24        4         4       16          0
Batch                        1         5        0         0        5          0
HTML                         1        38        6         0       32          0
License                      1        21        4         0       17          0
SVG                          1        61        0         0       61          0
Shell                        1         4        1         1        2          0
Smarty Template              1         6        2         0        4          0
───────────────────────────────────────────────────────────────────────────────
Total                      145    163578    18028     14183   131367      17988
───────────────────────────────────────────────────────────────────────────────
Estimated Cost to Develop (organic) $4,529,037
Estimated Schedule Effort (organic) 24.42 months
Estimated People Required (organic) 16.48
───────────────────────────────────────────────────────────────────────────────
Processed 5564571 bytes, 5.565 megabytes (SI)
───────────────────────────────────────────────────────────────────────────────
```
